ls1intum/Artemis
# `Development`: Improve course overview and course details performance

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
The course dashboard was loading way too much data and some pages (competencies or tutorial groups) were not displayed anymore or loaded too much / wrong data

### Description
Fix the above mentioned issues and reduce the database load for course overview and course detail related pages. There is further potential to optimized in follow-up PRs:
- Cache course icons
- Load less data in the `courses/for-dashboard` REST call
- Load data individually for each course detail page (exams, exercises, lectures, etc.) and do not load all data
- Improve the refresh button to actually only refresh data on this particular page and not load the whole course details again (which might not be useful on the tutorial group page)
- Inspect relevant DB queries for potential optimizations

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis
2. Navigate to Course Overview
3. Navigate into a Course and check that all pages are working

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Code Review
#### Manual Tests
